doc/direction.md

```diff
@@ -0,0 +1,52 @@
+# Minetest Direction Document
+
+
+
+## 0. Preface
+
+### 0.1. Introduction
+
+This document attempts to set out Minetest's philosophy, aims, and objectives.
+
+Please note that Minetest is developed by a community, and not everyone in the community
+holds the same views as to what Minetest should be. Therefore, defining a common direction is
+very difficult.
+
+### 0.2. Definitions
+
+* `Minetest` is the game engine.
+* `Minetest Game` is the default game which comes with the engine.
+* The `Minetest Project` refers to the combination of the above, the development team, and supporting tools.
+
+### 0.3. Other documents
+
+Please read celeron55's document: http://c55.me/blog/?p=1491
+
+Also see the roadmaps and manifestos of core developers on the forums: https://forum.minetest.net/viewforum.php?f=7
+
+
+## 1. Mission Statement
+
+Minetest provides both a game engine, which allows the creation of voxel games using Lua, and
+a content distribution platform. Players can connect to servers with their own game play, and
+easily install content through the Minetest client.
+
+## 2. Goals
+
+* **Stability and performance**, especially on older devices.
+* **Keep the engine lightweight**. Support a large amount of gameplay by pushing code and functionality out to Lua,
+  rather than having a complex core. Keeping the core as small as possible makes it
+  easier to test and more flexible.
+* **Keep the engine as universal as possible**. It's worth comparing everything that goes
+  into the engine for how well it suits a gravitation-less space game without ground surface.
+  If everything is either disableable or suitable for such, then it gives much more freedom
+  for people to experiment and create interesting things, even in ground-based worlds.
+* **Make polishing easy**. It shouldn't be made unnecessarily hard to make polished content
+  even while most content isn’t polished.
```

Comment:
Would you please be so nice to explain me what you mean by this sentence?

Reply:
it's from c55's roadmap

Reply:
Haha, this sounds like nobody knows what it means, but it has to be carried on for completeness. Maybe there's someone who understands this and who would blame you if you dropped this.

c55 uses "mustn't", "not shouldn't".

I think, what this wants to say is that there's much content (eg. mods code) that uses deprecated stuff, has bad formatting, wasn't updated for a long time and is buggy, but creating new content (eg. a new mod) should be easy to do correct, meaning that you should not need examples to make your own code or that you can easily read what features are deprecated.

Reply:
Sometimes, I picture celeron55 poppin onto the web and thinking... "OMFG, it's _still_ going!?"

The overall "direction" seems to be this way ------>
No, wait.... <-------- that way....
wait... give me a second...

Over the next year, MTG just needs to add some simple mobs (a few peaceful such as animals, and a couple aggressive) whether the pathfinding is perfected or not yet... and then add SmallJokers new bows mod, and players will be happy as a pig in...
...mud

This will keep the hordes at bay for a little while, while everyone figures out wtf to do, to put a solid direction/goals to focus on, whether it be an entirely new game to be packaged or what ever...
Even if it is similar to the current MTG but with a lot of current changes without all the patches & bondo covering the dents, causing bloat.

Reply:
> Would you please be so nice to explain me what you mean by this sentence?

I think it admits that most MT content isn't very 'polished', but warns against using that fact to justify making polished content excessively difficult to create.
Or maybe, it explains that it is not hypocritical to make polished content possible while most MT content isn't polished.
Makes sense to me.